presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestDistributedQueries.java

```diff
@@ -714,17 +721,45 @@ public void testQueryLoggingCount()
     {
         QueryManager queryManager = ((DistributedQueryRunner) getQueryRunner()).getCoordinator().getQueryManager();
         executeExclusively(() -> {
+            assertUntilTimeout(
+                    () -> assertEquals(
+                            queryManager.getAllQueryInfo()
+                                    .stream()
+                                    .filter(info -> !info.isFinalQueryInfo())
+                                    .collect(toList()),
+                            ImmutableList.of()),
+                    new Duration(1, MINUTES));
+
             long beforeQueryCount = queryManager.getStats().getCompletedQueries().getTotalCount();
             assertUpdate("CREATE TABLE test_query_logging_count AS SELECT 1 foo_1, 2 foo_2_4", 1);
             assertQuery("SELECT foo_1, foo_2_4 FROM test_query_logging_count", "SELECT 1, 2");
             assertUpdate("DROP TABLE test_query_logging_count");
             assertQueryFails("SELECT * FROM test_query_logging_count", ".*Table .* does not exist");
 
-            long queryCount = queryManager.getStats().getCompletedQueries().getTotalCount() - beforeQueryCount;
-            assertEquals(queryCount, 4, format("expected [%s] but found [%s]", 4, queryCount));
+            // TODO: Figure out a better way of synchronization
+            assertUntilTimeout(
+                    () -> assertEquals(queryManager.getStats().getCompletedQueries().getTotalCount() - beforeQueryCount, 4),
+                    new Duration(1, MINUTES));
         });
     }
 
+    private static void assertUntilTimeout(Runnable assertion, Duration timeout)
+    {
+        long start = System.nanoTime();
+        while (!Thread.currentThread().isInterrupted()) {
```

Comment:
And what if the thread _is_ interrupted? If the `assertUntilTimeout` was last instruction in the test, should the test complete successfully?